tests/bgp/test_traffic_shift.py

```diff
@@ -231,25 +231,47 @@ def verify_current_routes_announced_to_neighs(dut_host, neigh_hosts, orig_routes
     logger.info("Verifying all the original routes(ipv{}) are announced to bgp neighbors".format(ip_ver))
     cur_routes_on_all_nbrs.update(parse_routes_on_neighbors(dut_host, neigh_hosts, ip_ver))
     # Compare current routes after TSB with original routes advertised to neighbors
-    if cur_routes_on_all_nbrs != orig_routes_on_all_nbrs:        
+    if cur_routes_on_all_nbrs != orig_routes_on_all_nbrs:     
         return False           
     return True
 
-def log_missing_routes(orig_routes_on_all_nbrs, cur_routes_on_all_nbrs):
+def check_and_log_routes_diff(duthost, neigh_hosts, orig_routes_on_all_nbrs, cur_routes_on_all_nbrs,ip_ver):
     cur_nbrs = set(cur_routes_on_all_nbrs.keys())
     orig_nbrs = set(orig_routes_on_all_nbrs.keys())
     if cur_nbrs != orig_nbrs:
         logger.warn("Neighbor list mismatch: {}".format(cur_nbrs ^ orig_nbrs))
         return False
-
+    
+    routes_dut = parse_rib(duthost, ip_ver)
     for hostname in orig_routes_on_all_nbrs.keys():
```

Comment:
I didn't see why we need embedded for loop here @tjchadaga

Reply:
this was added by mistake during cleanup. Thanks for flagging!